### PR TITLE
database performance audit

### PR DESCRIPTION
**Pull request by GhostEsso**
Hello **Code Reviewer** :smiley: 
We hope you are doing well.
In this Milestone, my partner @Nelofarzabi  and I worked on the project requirements:
- We  Improved database performance
- Added email column to owners
- Indexes created

the screenshots are below

**Demo before audit**

Capture d’écran du 2023-09-25 12-56-36

Capture d’écran du 2023-09-25 12-57-45

Capture d’écran du 2023-09-25 12-47-16

**After demo audit**

Capture d’écran du 2023-09-25 15-27-59

Capture d’écran du 2023-09-25 15-29-20

Capture d’écran du 2023-09-25 15-28-36

Thanks for reviewing our project :pray:

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -55,9 +55,43 @@ CREATE TABLE specializations (
...
+CREATE TABLE visits(
+  id INT GENERATED ALWAYS AS IDENTITY,
+  animal_id INT REFERENCES animals(id),
+  vet_id INT REFERENCES vets(id),
+  date_of_visits DATE,
+  PRIMARY KEY(id)
+);
+
+CREATE INDEX vet_id_index ON visits (vet_id DESC);
+CREATE INDEX animal_id_index ON visits (animal_id);
```
Comment (shahadat3669) [inline schema.sql:86]: - [ ] 👍 Great work on the project! While your vet_id_index and owners_index screenshots look solid, there's a minor issue with the screenshot for animal_id_index. It appears that both screenshots of animal_id were taken after adding the index. To better represent the state and provide a complete comparison, kindly consider taking a screenshot of the animal_id query execution plan before adding the index. You can achieve this by removing the indexes temporarily, running the query, capturing the `explain analyze` output, and then adding the indexes back.

For reference, you can check out an example [here](https://github.com/errea/vet_clinic_database/pull/1#pullrequestreview-763088027). Keep up the excellent work! 👍🔍📸

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @GhostEsso, @Nelofarzabi, 

Great job so far! 👏 You've done well, and you're almost there for the next project ⏭️, but there are a few issues 🐛 that you still need to work on.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights

✅ Everyone has contributed 
✅ Used `EXPLAIN ANALYZE` command for each
✅ `INSERT INTO` queries are saved in the data.sql file

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (GhostEsso) [reply schema.sql:86]: @shahadat3669 Thank you for reviewing our code. Thank you also for guiding me through your request. I took a screenshot of the execution plan for the animal_id request before adding the index but by putting the screenshots in the pull request I was wrong. I have provided the correct screenshot, once again thank you for your time and guidance :pray:

**Review by TuyishimireEric — APPROVED** [review]
# Status: APPROVED 🟢  
### Hi  @GhostEsso   👋 👋 

#### I'm @TuyishimireEric, Your code reviewer,
Good job on implementing the project requirements! 🥇 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it 🦾  
Congratulations! 🎉


## Highlight 👍 
✅ used `explain analyze` to detect and fix long-lasting queries.


### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/enus/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**